File: C++/63-数据流中的中位数.cpp
class Solution
{
public:
    vector<int> data;
    void Insert(int num)
    {
        data.push_back(num);
        sort(data.begin(), data.end());
    }

    double GetMedian()
    {
        int n = data.size();
        if (n & 1)
            return data[n / 2];
        else
            return 0.5 * (data[n / 2 - 1] + data[n / 2]);
    }
};

File: Python/63-数据流中的中位数.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        self.data = []
    def Insert(self, num):
        # write code here
        if not self.data:
            self.data.append(num)
        else:
            i = 0
            while i < len(self.data):
                if num <= self.data[i]:
                    break
                i += 1
            self.data.insert(i, num)
    def GetMedian(self, _):
        # write code here
        n = len(self.data)
        if n & 1 == 1:
            return self.data[n / 2]
        else:
            return 0.5 * (self.data[n / 2] + self.data[n / 2 - 1])